### PR TITLE
Port analyzers to Nomad

### PR DESCRIPTION
**Pull request by twunderlich-grapl**
### Which issue does this PR correspond to?
https://github.com/grapl-security/issue-tracker/issues/602

### What changes does this PR make to Grapl? Why?
This ports over the analyzers to Nomad. Note that this is still using the docker network hack for the moment

### How were these changes tested?
Follow the nomad instructions in BUILDING.md
Then go to the nomad UI and check that both containers run without failing AND check the logs for any errors

Note: analyzer-executor doesn't is trying to retrieve from an empty queue. Presumably once we start feeding data into that queue we'll see actual info in the logs

### CODE REVIEW

File: nomad/grapl-core.nomad [hunk 1]
```diff
@@ -196,6 +243,11 @@ locals {
 
   # String that contains all of the running Alphas for clients connecting to Dgraph (so they can do loadbalancing)
   alpha_grpc_connect_str = join(",", [for alpha in local.dgraph_alphas : "localhost:${alpha.grpc_public_port}"])
+
+  redis_trimmed = trimprefix(var.redis_endpoint, "redis://")
+  redis         = split(":", local.redis_trimmed)
+  redis_host    = local.redis[0]
+  redis_port    = local.redis[1]
```
Comment (wimax-grapl) [inline nomad/grapl-core.nomad:250]: nice!!

File: nomad/grapl-core.nomad [hunk 2]
```diff
@@ -578,6 +630,71 @@ job "grapl-core" {
...
+      env {
+        # AWS vars
+        AWS_REGION                  = var.aws_region
+        GRAPL_AWS_ACCESS_KEY_ID     = var.aws_access_key_id
+        GRAPL_AWS_ACCESS_KEY_SECRET = var.aws_access_key_secret
+        GRAPL_AWS_ENDPOINT          = var.aws_endpoint
+        # rust vars
+        RUST_LOG       = var.rust_log
+        RUST_BACKTRACE = 1
+        # service vars
```
Comment (wimax-grapl) [inline nomad/grapl-core.nomad:652]: I like this sectioning stuff you're doing

File: nomad/grapl-core.nomad [hunk 3]
```diff
@@ -578,6 +630,71 @@ job "grapl-core" {
...
+      env {
+        # AWS vars
+        AWS_REGION                  = var.aws_region
+        GRAPL_AWS_ACCESS_KEY_ID     = var.aws_access_key_id
+        GRAPL_AWS_ACCESS_KEY_SECRET = var.aws_access_key_secret
+        GRAPL_AWS_ENDPOINT          = var.aws_endpoint
+        # python vars
+        GRAPL_LOG_LEVEL = "INFO"
+        # presumably not needed but will confirm after
+        RUST_LOG = "INFO"
```
Comment (wimax-grapl) [inline nomad/grapl-core.nomad:680]: a) It's not needed, a-e is pure python for now.
b) i also hate that we have two disparate log level variabels and I want to fix it in the future - but that's a problem for Future Grapl

File: nomad/grapl-core.nomad [hunk 4]
```diff
@@ -578,6 +630,71 @@ job "grapl-core" {
...
+        # dgraph vars
+        MG_ALPHAS = local.alpha_grpc_connect_str
+        # service vars
+        GRAPL_ANALYZER_MATCHED_SUBGRAPHS_BUCKET = var.analyzer_matched_subgraphs_bucket
+        GRAPL_ANALYZERS_BUCKET                  = var.analyzer_bucket
+        GRAPL_MODEL_PLUGINS_BUCKET              = var.model_plugins_bucket
+        SOURCE_QUEUE_URL                        = var.analyzer_executor_queue
+        GRPC_ENABLE_FORK_SUPPORT                = "1"
```
Comment (wimax-grapl) [inline nomad/grapl-core.nomad:688]: While I recognize that in the docker-compose it's a string - i think it's fine to leave it as an int. doesn't really matter afaict

File: pulumi/grapl/__main__.py [hunk 5]
```diff
@@ -136,11 +141,19 @@ def main() -> None:
         kafka = Kafka("kafka")
 
         job_vars = pulumi.Output.all(
+            analyzer_bucket=analyzers_bucket.bucket,
+            # This should be dispatched-analyzer-bucket, which doesn't appear to exist?
+            analyzer_dispatched_bucket=analyzers_bucket.bucket,
```
Comment (wimax-grapl) [inline pulumi/grapl/__main__.py:146]: I found it - I was as confused as you were until I realized it was doing string interpolation to construct "dispatched-analyzer-bucket"
```
    dispatched_analyzer_emitter = emitter.EventEmitter("dispatched-analyzer")
````
and in EventEmitter:
```
        bucket_name = f"{event_name}-bucket"
        pulumi.export(bucket_name, self.bucket.bucket)
```

so you probably want `dispatched_analyzer_emitter.bucket.bucket` ?

Comment (twunderlich-grapl) [reply pulumi/grapl/__main__.py:146]: Ah yes. I think I actually did this with a different bucket too 🤦 . Fixing now

**Review by wimax-grapl — APPROVED** [review]

**Pull request merged**